utils/swift_build_support/swift_build_support/products/ninja.py

```diff
@@ -28,6 +28,18 @@ class Ninja(product.Product):
     def is_build_script_impl_product(cls):
         return False
 
+    @classmethod
+    def make_builder(cls, args, toolchain, workspace, host):
+        return NinjaBuilder(args, toolchain, workspace)
+
+
+class NinjaBuilder(product_builder.ProductBuilder):
+    def __init__(self, args, toolchain, workspace):
```

Comment:
I know you're mostly moving code around here, but it seems odd to me that we're passing around the entire argument 'Namespace' object rather than just the `darwin_deployment_version_osx`.

Reply:
I have been trying to keep the interface for the builders almost the same for all of them. I need a little bit of consistency in `make_builder` because how it is invoked, and I keep the same arguments in the builders to keep them similar.

It is true that this one might only use one of the arguments, but other builders use many arguments, so I will be prefer to keep all the arguments together, just in case in the future some more arguments are needed (which might not be the case for Ninja, but I'm pretty sure will be something that will happen frequently in Swift).

Reply:
@Rostepher - I have to agree with @drodriguez here - passing around the additional information is worth it because it means all the invocations are the same.  We should push that down into the ABC here (that is, require that from the `product_builder.ProductBuilder`.  That makes it clearer that the reason that the arguments are being passed this way is not because this particular builder requires it but because it is the "generic" interface.  This would be extremely helpful as the builders start building more complex products.

Reply:
I didn’t want to push the constructor into the ABC. The reason is that when the builder that uses `build-script-impl` gets implemented, there's one parameter (the product being build) which the ABC has no business knowing. This will also happen when the builder that uses CMake is implemented (since both are builders that can build different products). In those two cases, the arguments as a whole are passed because one doesn't know what each particular product will use or not.

I will prefer to to impose constructor restrictions in the base class, but if people disagree, I might have been able to make it work.

Reply:
Hmm, I don't think I have a strong objection to it.  Just seems that any product builder should know what it is building.